src/coreclr/jit/codegenlinear.cpp

```diff
@@ -2484,8 +2484,8 @@ CodeGen::GenIntCastDesc::GenIntCastDesc(GenTreeCast* cast)
         }
 
 #if defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
-        // For LoongArch64's ISA which is same with the MIPS64 ISA, even the instructions of 32bits operation need
-        // the upper 32bits be sign-extended to 64 bits.
+        // For LoongArch64's ISA which is a RISC ISA which is different from any other existing ones,
+        // even the instructions of 32bits operation need the upper 32bits be sign-extended to 64 bits.
```

Comment:
```suggestion
        // LoongArch64 and Riscv64 ABI require 32-bit values to be sign extended to 64-bits.
        // We apply the sign-extension unconditionally here to avoid corner case bugs, even
        // though it may not be strictly necessary in all cases.
```
How about this?

Reply:
The ABI only requires the values to be sign-extended to 64-bits for function arguments and return values, not everywhere.

Reply:
Right, it is what I have tried to capture in the suggestion.

This code looks like a quick workaround. If this was done properly, this sign extension would be applied only when it is actually necessary.

Reply:
Thanks